router: Data race in proxyproto tests

From CI build https://ci.flynn.io/builds/20161118152552-6ac314c3:

```
==================
WARNING: DATA RACE
Read at 0x00c42009a284 by goroutine 13:
  testing.(*common).Fail()
      /home/ubuntu/go/src/github.com/flynn/flynn/util/_toolchain/go/src/testing/testing.go:411 +0x9f
  testing.(*common).FailNow()
      /home/ubuntu/go/src/github.com/flynn/flynn/util/_toolchain/go/src/testing/testing.go:431 +0x38
  testing.(*common).Fatalf()
      /home/ubuntu/go/src/github.com/flynn/flynn/util/_toolchain/go/src/testing/testing.go:496 +0x94
  github.com/flynn/flynn/router/proxyproto.TestParse_BadHeader.func1()
      /home/ubuntu/go/src/github.com/flynn/flynn/router/proxyproto/protocol_test.go:198 +0x2e2

Previous write at 0x00c42009a284 by goroutine 12:
  testing.tRunner.func1()
      /home/ubuntu/go/src/github.com/flynn/flynn/util/_toolchain/go/src/testing/testing.go:605 +0x34a
  testing.tRunner()
      /home/ubuntu/go/src/github.com/flynn/flynn/util/_toolchain/go/src/testing/testing.go:612 +0xe5

Goroutine 13 (running) created at:
  github.com/flynn/flynn/router/proxyproto.TestParse_BadHeader()
      /home/ubuntu/go/src/github.com/flynn/flynn/router/proxyproto/protocol_test.go:200 +0x16c
  testing.tRunner()
      /home/ubuntu/go/src/github.com/flynn/flynn/util/_toolchain/go/src/testing/testing.go:610 +0xc9

Goroutine 12 (finished) created at:
  testing.(*T).Run()
      /home/ubuntu/go/src/github.com/flynn/flynn/util/_toolchain/go/src/testing/testing.go:646 +0x52f
  testing.RunTests.func1()
      /home/ubuntu/go/src/github.com/flynn/flynn/util/_toolchain/go/src/testing/testing.go:793 +0xb9
  testing.tRunner()
      /home/ubuntu/go/src/github.com/flynn/flynn/util/_toolchain/go/src/testing/testing.go:610 +0xc9
  testing.RunTests()
      /home/ubuntu/go/src/github.com/flynn/flynn/util/_toolchain/go/src/testing/testing.go:799 +0x4ba
  testing.(*M).Run()
      /home/ubuntu/go/src/github.com/flynn/flynn/util/_toolchain/go/src/testing/testing.go:743 +0x12f
  main.main()
      github.com/flynn/flynn/router/proxyproto/_test/_testmain.go:106 +0x2f8
==================
panic: Fail in goroutine after TestParse_BadHeader has completed

goroutine 21 [running]:
panic(0x5c9a80, 0xc420098250)
        /home/ubuntu/go/src/github.com/flynn/flynn/util/_toolchain/go/src/runtime/panic.go:500 +0x1ae
testing.(*common).Fail(0xc42009a240)
        /home/ubuntu/go/src/github.com/flynn/flynn/util/_toolchain/go/src/testing/testing.go:412 +0x182
testing.(*common).FailNow(0xc42009a240)
        /home/ubuntu/go/src/github.com/flynn/flynn/util/_toolchain/go/src/testing/testing.go:431 +0x39
testing.(*common).Fatalf(0xc42009a240, 0x6002c6, 0x7, 0xc42003af70, 0x1, 0x1)
        /home/ubuntu/go/src/github.com/flynn/flynn/util/_toolchain/go/src/testing/testing.go:496 +0x95
github.com/flynn/flynn/router/proxyproto.TestParse_BadHeader.func1(0xc420098140, 0xc42009a240)
        /home/ubuntu/go/src/github.com/flynn/flynn/router/proxyproto/protocol_test.go:198 +0x2e3
created by github.com/flynn/flynn/router/proxyproto.TestParse_BadHeader
        /home/ubuntu/go/src/github.com/flynn/flynn/router/proxyproto/protocol_test.go:200 +0x16d
FAIL    github.com/flynn/flynn/router/proxyproto        0.045s
```

--- a/router/proxyproto/protocol_test.go
+++ b/router/proxyproto/protocol_test.go
@@ -2,7 +2,6 @@ package proxyproto
 
 import (
 	"bytes"
-	"io"
 	"net"
 	"testing"
 )
@@ -18,7 +17,8 @@ func TestPassthrough(t *testing.T) {
 	go func() {
 		conn, err := net.Dial("tcp", pl.Addr().String())
 		if err != nil {
-			t.Fatalf("err: %v", err)
+			t.Errorf("err: %v", err)
+			return
 		}
 		defer conn.Close()
 
@@ -26,10 +26,12 @@ func TestPassthrough(t *testing.T) {
 		recv := make([]byte, 4)
 		_, err = conn.Read(recv)
 		if err != nil {
-			t.Fatalf("err: %v", err)
+			t.Errorf("err: %v", err)
+			return
 		}
 		if !bytes.Equal(recv, []byte("pong")) {
-			t.Fatalf("bad: %v", recv)
+			t.Errorf("bad: %v", recv)
+			return
 		}
 	}()
 
@@ -64,7 +66,8 @@ func TestParse_ipv4(t *testing.T) {
 	go func() {
 		conn, err := net.Dial("tcp", pl.Addr().String())
 		if err != nil {
-			t.Fatalf("err: %v", err)
+			t.Errorf("err: %v", err)
+			return
 		}
 		defer conn.Close()
 
@@ -76,10 +79,12 @@ func TestParse_ipv4(t *testing.T) {
 		recv := make([]byte, 4)
 		_, err = conn.Read(recv)
 		if err != nil {
-			t.Fatalf("err: %v", err)
+			t.Errorf("err: %v", err)
+			return
 		}
 		if !bytes.Equal(recv, []byte("pong")) {
-			t.Fatalf("bad: %v", recv)
+			t.Errorf("bad: %v", recv)
+			return
 		}
 	}()
 
@@ -123,7 +128,8 @@ func TestParse_ipv6(t *testing.T) {
 	go func() {
 		conn, err := net.Dial("tcp", pl.Addr().String())
 		if err != nil {
-			t.Fatalf("err: %v", err)
+			t.Errorf("err: %v", err)
+			return
 		}
 		defer conn.Close()
 
@@ -135,10 +141,12 @@ func TestParse_ipv6(t *testing.T) {
 		recv := make([]byte, 4)
 		_, err = conn.Read(recv)
 		if err != nil {
-			t.Fatalf("err: %v", err)
+			t.Errorf("err: %v", err)
+			return
 		}
 		if !bytes.Equal(recv, []byte("pong")) {
-			t.Fatalf("bad: %v", recv)
+			t.Errorf("bad: %v", recv)
+			return
 		}
 	}()
 
@@ -182,21 +190,12 @@ func TestParse_BadHeader(t *testing.T) {
 	go func() {
 		conn, err := net.Dial("tcp", pl.Addr().String())
 		if err != nil {
-			t.Fatalf("err: %v", err)
+			t.Errorf("err: %v", err)
+			return
 		}
 		defer conn.Close()
 
-		// Write out the header!
-		header := "PROXY TCP4 what 127.0.0.1 1000 2000\r\n"
-		conn.Write([]byte(header))
-
-		conn.Write([]byte("ping"))
-
-		recv := make([]byte, 4)
-		_, err = conn.Read(recv)
-		if err != io.EOF {
-			t.Fatalf("err: %v", err)
-		}
+		conn.Write([]byte("PROXY TCP4 what 127.0.0.1 1000 2000\r\nping"))
 	}()
 
 	conn, err := pl.Accept()

